### PR TITLE
fix(ui) Display duplicate nodes in lineage viz

### PR DESCRIPTION
**Pull request by chriscollins3456**
In a previous attempt to fix a bug where switching base nodes caused weird duplicates and wonky dragging behavior, we prevented the lineage graph from rendering any duplicates. Turns out some people want to render duplicates in their graph because a singular node could be upstream and downstream of a given entity. This reverts the change to allow duplicates and fixes the base node switching bug by making the `key` on each node unique. I also clean things up a bit and remove an unnecessary `direction` prop being passed.

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-web-react/src/app/lineage/LineageTreeNodeAndEdgeRenderer.tsx [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { curveBasis } from '@vx/curve';
 import { LinePath } from '@vx/shape';
 import { TransformMatrix } from '@vx/zoom/lib/types';
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/lineage/LineageTreeNodeAndEdgeRenderer.tsx:5]: Any component with an 'And' in the name makes me sad

File: datahub-web-react/src/app/lineage/LineageTreeNodeAndEdgeRenderer.tsx [hunk 2]
```diff
@@ -91,18 +89,19 @@ export default function LineageTreeNodeAndEdgeRenderer({
...
-            {nodesToRender.map((node) => {
+            {nodesToRender.map((node, index) => {
                 const isSelected = node.data.urn === selectedEntity?.urn;
                 const isHovered = node.data.urn === hoveredEntity?.urn;
+                const key = `node-${node.data.urn}-${node.direction}-${index}`;
 
                 return (
                     <LineageEntityNode
-                        key={`node-${node.data.urn}-${direction}`}
+                        key={key}
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/lineage/LineageTreeNodeAndEdgeRenderer.tsx:104]: Not sure if this is relevant, but is there a way to use a triplet to render the node?

node-node.data.urn-dest.data.urn-direction?

Or is this not really the problem

Comment (chriscollins3456) [reply datahub-web-react/src/app/lineage/LineageTreeNodeAndEdgeRenderer.tsx:104]: yeah so the problem when switching base nodes is we had keys here that weren't unique and we would leave the old base node rendered.

but as far as the general issue of having duplicate nodes and dragging and what not, having some combo like that could certainly work. idk if it's relevant in this specific situation though

File: datahub-web-react/src/app/lineage/LineageTreeNodeAndEdgeRenderer.tsx (hunk 1 above)
Comment (chriscollins3456) [reply datahub-web-react/src/app/lineage/LineageTreeNodeAndEdgeRenderer.tsx:5]: yes lol

**Review by jjoyce0510 — APPROVED** [review]

**Pull request merged**